libraries/Microsoft.Bot.Connector/OAuthClientConfig.cs

```diff
@@ -0,0 +1,120 @@
+
+        /// <summary>
+        /// Send a dummy OAuth card when the bot is being used on the Emulator for testing without fetching a real token.
+        /// </summary>
+        /// <param name="emulateOAuthCards">Indicates whether the Emulator should emulate the OAuth card.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        public static async Task SendEmulateOAuthCardsAsync(OAuthClient client, bool emulateOAuthCards)
+        {
+            // Tracing
+            bool _shouldTrace = ServiceClientTracing.IsEnabled;
+            string _invocationId = null;
+            if (_shouldTrace)
+            {
+                _invocationId = ServiceClientTracing.NextInvocationId.ToString();
+                Dictionary<string, object> tracingParameters = new Dictionary<string, object>();
+                tracingParameters.Add("emulateOAuthCards", emulateOAuthCards);
+                ServiceClientTracing.Enter(_invocationId, client, "GetToken", tracingParameters);
+            }
+            // Construct URL
+            var _baseUrl = client.BaseUri.AbsoluteUri;
+            var _url = new Uri(new Uri(_baseUrl + (_baseUrl.EndsWith("/") ? "" : "/")), "api/usertoken/emulateOAuthCards?emulate={emulate}").ToString();
+            _url = _url.Replace("{emulate}", emulateOAuthCards.ToString());
+
+            // Create HTTP transport objects
+            var _httpRequest = new HttpRequestMessage();
+            HttpResponseMessage _httpResponse = null;
+            _httpRequest.Method = new HttpMethod("POST");
+            _httpRequest.RequestUri = new System.Uri(_url);
+
+            var cancellationToken = CancellationToken.None;
+
+            // Serialize Request
+            string _requestContent = null;
+            // Set Credentials
+            if (client.Credentials != null)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                await client.Credentials.ProcessHttpRequestAsync(_httpRequest, cancellationToken).ConfigureAwait(false);
+            }
+            // Send Request
+            if (_shouldTrace)
+            {
+                ServiceClientTracing.SendRequest(_invocationId, _httpRequest);
+            }
+            cancellationToken.ThrowIfCancellationRequested();
+            _httpResponse = await client.HttpClient.SendAsync(_httpRequest, cancellationToken).ConfigureAwait(false);
+            if (_shouldTrace)
+            {
+                ServiceClientTracing.ReceiveResponse(_invocationId, _httpResponse);
+            }
+            HttpStatusCode _statusCode = _httpResponse.StatusCode;
+            cancellationToken.ThrowIfCancellationRequested();
+            string _responseContent = null;
+            if ((int)_statusCode != 200 && (int)_statusCode != 404)
+            {
+                var ex = new ErrorResponseException(string.Format("Operation returned an invalid status code '{0}'", _statusCode));
```

Comment:
This variable name "ex" is confusing. Usually I only see ex in "catch (MyExceptoin ex)" scenarios.